src/app/view/sharing/SharingProvider.tsx

```diff
@@ -41,10 +54,31 @@ interface SharingProviderProps {
 export const SharingProvider = ({
   children
 }: SharingProviderProps): JSX.Element => {
+  const client = useClient()
+  const session = useSession()
   const [state, dispatch] = useReducer(sharingReducer, {
     sharingIntentStatus: SharingIntentStatus.Undetermined,
-    filesToUpload: []
+    filesToUpload: [],
+    routeToUpload: undefined
   })
+  const { data: fetchedApps } = useQuery(
+    fetchSharingCozyApps.definition,
+    fetchSharingCozyApps.options
+  )
+
+  useEffect(() => {
+    if (client === null) return
+
+    const route = getRouteToUpload(
+      fetchedApps as SharingCozyApp[],
```

Comment:
Are we ok to call this `getRouteToUpload` with a `fetchedApps` equals to null because fetchStatus===loading/pending/errored  ?

Reply:
In my mind yes because it defers the null check to the low-level function (it will handle null or empty array data). That way the `SharingProvider` (to be renamed as we said) can focus on the user flow logic without too much coupling to data handling

Reply:
if you say so, but you have to deal with the fact that the request can fail (fetchStatus===errored) somewhere.

Reply:
> if you say so, but you have to deal with the fact that the request can fail (fetchStatus===errored) somewhere.

true, I didn't test for this scenario.

- [ ] To update